autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/update/ActivityUpdateEvaluator.java

```diff
@@ -0,0 +1,102 @@
+
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_FAILED;
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_IN_PROGRESS;
+import static com.sequenceiq.periscope.api.model.ActivityStatus.SCALING_FLOW_SUCCESS;
+import static java.util.Objects.isNull;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.flow.api.model.FlowCheckResponse;
+import com.sequenceiq.periscope.domain.ScalingActivity;
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.context.EvaluatorContext;
+import com.sequenceiq.periscope.monitor.context.ScalingActivitiesEvaluatorContext;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.handler.FlowCommunicator;
+import com.sequenceiq.periscope.service.ScalingActivityService;
+
+@Component("ActivityUpdateEvaluator")
+@Scope("prototype")
+public class ActivityUpdateEvaluator extends EvaluatorExecutor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActivityUpdateEvaluator.class);
+
+    private static final String EVALUATOR_NAME = ActivityUpdateEvaluator.class.getName();
+
+    @Inject
+    private FlowCommunicator flowCommunicator;
+
+    @Inject
+    private ScalingActivityService scalingActivityService;
+
+    private ScalingActivities scalingActivities;
+
+    @Nonnull
+    @Override
+    public EvaluatorContext getContext() {
+        return new ScalingActivitiesEvaluatorContext(scalingActivities);
+    }
+
+    @Override
+    public String getName() {
+        return EVALUATOR_NAME;
+    }
+
+    @Override
+    public void setContext(EvaluatorContext context) {
+        this.scalingActivities = (ScalingActivities) context.getData();
+    }
+
+    @Override
+    protected void execute() {
+        if (isNull(scalingActivities)) {
+            LOGGER.info("No ScalingActivities found");
+            return;
+        }
+
+        LOGGER.info("Executing ScalingActivityUpdateEvaluator: {}", scalingActivities.getId());
+
+        try {
+            if (scalingActivities.getActivityIds().isEmpty()) {
+                LOGGER.info("No activityIds present, skipping request to update");
+            } else {
+                Map<String, Long> activityIdsByFlowIds = scalingActivityService.findAllByIds(scalingActivities.getActivityIds()).stream()
+                        .collect(toMap(ScalingActivity::getFlowId, ScalingActivity::getId));
+                Map<Long, FlowCheckResponse> flowResponseByActivityId = flowCommunicator.getFlowStatusFromFlowIds(activityIdsByFlowIds);
```

Comment:
flowCommunicator seems to request 50 events. In the case this list is larger than 50 - should we be looping over here?
That's likely better than waiting for the next execution
1) we could end up building a backlog if each run is artificially limited to 50,
2) potentially delays when an event is updated.

(Should have a unit test to cover this scenario as well)

Reply:
Added logic to loop over to the remainder of flowIds in case the input list is larger than 50 in `FlowCommunicator`. Covered this scenario with UTs.

Reply:
I don't see this: here on in FlowCommunicator - what am I missing ?

Reply:
In `FlowCommunicator`, I created a concurrentHasSet that stores the remainder of flowIds from the previous invocation (that weren't queried to CB), and merged that with the incoming list of flowIds.

Reply:
So, there will always be a waitlist? can this be done using a loop like Sid said? if the invocations are done every 10 mins, you will not have the complete results of the first invocation until the next run or more.

Reply:
Updated this to loop over until the entire input list is processed. Had concerns regarding this loop because it would lead to multiple API calls